src/components/Firework/Firework.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {FC, useEffect, useRef, useState} from 'react'
+import {Dimensions, View, Animated, Easing} from 'react-native'
+import {getRandomNumber} from './utils'
+
+import styles from './styles'
+
+type FireworkProps = {
+  density?: number
+  iterations?: number
+  duration?: number
+}
+type CoordsType = {
+  x: number[]
+  y: number[]
+}
+
+export const Firework: FC<FireworkProps> = ({
+  density = 5,
+  iterations = 30,
+  duration = 700,
+}) => {
+  const [coords, setCoords] = useState<CoordsType>({
+    x: [],
+    y: [],
+  })
+  const opacityAnim = useRef(new Animated.Value(1)).current
+  const moveAnim = useRef(new Animated.Value(0)).current
```

Comment:
overall looks good, I might be mistaken, but you can use only one animation value, and then for opacity you can use interpolation with more that one inputRange

for example
moveAnim goes from 0 to 1
then if you want opacity to increase on start and then fade on end you can use

const opaicty = moveAnim.interpolate({
  inputRange: [0, 0.15, 0.5, 0.9, 1],
  outputRange: [0, 1, 1, 1, 0],
  extrapolate: 'clamp',
})

Reply:
Yeah, I did interpolation for animation below as well

```
let ballOpacity = opacityAnim.interpolate({
      inputRange: [0, 1],
      outputRange: [0, 1],
    })
```
But maybe I did get the point properly.

Reply:
I mean that you can avoid having opacityAnim and moveAnim together. You can connect your opacity styling directly to moveAnim.
Controlling and coordinate only one values is much easier than many

Reply:
ah, I see, thanks